generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -802,4 +804,60 @@ public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
 
   }
 
+  private class IdentityRewriteComposite extends AbstractIdentityTransformation {
+    private IdentityRewriteComposite() {
+      // all non-boolean vector/matrix types
+      super(BasicType.allNumericTypes().stream().filter(
+          item -> !BasicType.allScalarTypes().contains(item)).collect(Collectors.toList()),
+          false);
+    }
+
+    @Override
+    public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
+                      Fuzzer fuzzer) {
+      // v -> (true ? vecX(..., identity(v[Y]), ...) : _)
+      // v -> (false ? _ : vecX(..., identity(v[Y]), ...))
+      // where v is a vector of size X, y is the random entry in v we want to apply
+      // identities to, and ... is the other entries in v that we don't change.
+      // Similarly for matrices.
+
+      assert BasicType.allVectorTypes().contains(type)
+          || BasicType.allMatrixTypes().contains(type);
+
+      final int numColumns =
```

Comment:
Even though it is clunky, I would favour "numElementsOrColumns" as a name here, because in the case of a vector we call them elements.

But actually, how about "numIndices"?  That would be nice as you later have a variable called "indexToApplyIdentities".